src/TypeSortedCollections.jl

```diff
@@ -220,6 +220,22 @@ end
     end
 end
 
+@generated function Base.any(f::F, tsc::TypeSortedCollection{<:Any, N}) where {F, N}
+    N == 0 && return false
+    quote
+        Base.@_inline_meta
+        $(foldr((x, y) -> :($x || $y), [:(any(f, tsc.data[$i])) for i in 1 : N]))
```

Comment:
This doesn't handle `missing`, which `Base.any` does. I'm OK with not handling that case right now, but could you add a comment?

Reply:
I thought that `&&` and `||` do work with missing, but order is important it seems
```julia
julia> missing || false
ERROR: TypeError: non-boolean (Missing) used in boolean context
Stacktrace:
 [1] top-level scope at none:0

julia> false || missing
missing
```
Seeding the reduction with `false` should make this work for `missing` (nothing about TSC sorting should introduce any problems, I think) and eliminate the base case above.

Reply:
Actually, this is way trickier than I thought -- I don't think a one-liner will suffice. I'll unroll the `Base.any` for loop.